### ISSUE TITLE
JSON endpoint for now playing

### ISSUE THREAD

**Issue by case** [overwatchcorp/kdfc-playlist-bot#1]
Tl;dr: I think this endpoint -- https://schedule.kdfc.com/now/KDFC.json -- has everything needed to keep the bot more in sync (and efficient) with the KDFC stream.

Hi Jade,

Thanks for making this bot! I use it all the time to keep tabs on what KDFC is currently playing. I've been meaning to create a PR for this, but haven't been able to make the time unfortunately. So I figured I'd file what I've found so far, in case anyone else is interested in looking into it.

I've noticed a few times that the Twitter account gets out of sync with what's on the radio, and I was looking into alternate ways to retrieve this metadata (vs. scraping the whole web page every 10s). 

One idea was to stream the audio via a server instance somewhere, and pull out the metadata. e.g. via https://github.com/gabek/node-internet-radio -- but this doesn't include the rich metadata that we see on the KDFC website. For example, `Florence Price - Symphony #1 in e` is what appears via the audio stream, but nothing else.

I noticed their JSON endpoint -- https://schedule.kdfc.com/now/KDFC.json -- while looking through Chrome DevTools, and I think it has all the relevant metadata:

- `"start"` has a timestamp for when the track started
- `"end"` has a timestamp for when the track will end, so we'd know when to check the JSON again.
- For the music, there's `title`, `artist`, `Composer`, `Conductor`, `Orchestra`, etc.

**Comment by overwatchcorp** [comment]
hi i just saw this! i think i have some time and i'll look in to implementing this

**Comment by case** [comment]
And fwiw, I grabbed a few json blobs for tests:

Soloist = empty string:

```json
{
  "id": "ae0j9knqvla2gr410pbpk5v40s",
  "created": "2020-11-04T02:41:11.000Z",
  "updated": "2020-11-04T02:41:11.585Z",
  "summary": "Symphony #30 in D K 202 by Wolfgang Amadeus Mozart \nSinfonia Varsovia Emmanuel Krivine  \nDenon 79784",
  "description": "sched_time: 67451600\nair_time: 67267000\nstack_pos: \ntitle: Symphony #30 in D K 202\nartist: Wolfgang Amadeus Mozart\ntrivia: \ncategory: MUA\ncart: 06F7\nintro: 0\nend: \nstation: KDFC-FM\nduration: 1092649\nmedia_type: SONG\nmilliseconds_left: \nComposer: Wolfgang Amadeus Mozart\nComposer_Last: Mozart, Wolfgang Amadeus\nConductor: Emmanuel Krivine\nISRC: \nMM_Song_ID: 22973\nOrchestra: Sinfonia Varsovia\nRecord_Company: Denon\nSoloist: \nSpine_Number: 79784\n",
  "start": {
    "dateTime": "2020-11-03T18:41:07-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "end": {
    "dateTime": "2020-11-03T18:59:19-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "extraInfo": {
    "sched_time": "67451600",
    "air_time": "67267000",
    "stack_pos": "",
    "title": "Symphony #30 in D K 202",
    "artist": "Wolfgang Amadeus Mozart",
    "trivia": "",
    "category": "MUA",
    "cart": "06F7",
    "intro": "0",
    "end": "",
    "station": "KDFC-FM",
    "duration": "1092649",
    "media_type": "SONG",
    "milliseconds_left": "",
    "Composer": "Wolfgang Amadeus Mozart",
    "Composer_Last": "Mozart, Wolfgang Amadeus",
    "Conductor": "Emmanuel Krivine",
    "ISRC": "",
    "MM_Song_ID": "22973",
    "Orchestra": "Sinfonia Varsovia",
    "Record_Company": "Denon",
    "Soloist": "",
    "Spine_Number": "79784"
  }
}
```

Soloist has a value:

```json
{
  "id": "0duquqq5fqj5ml3dr5g2q529oo",
  "created": "2020-11-04T03:11:18.000Z",
  "updated": "2020-11-04T03:11:18.699Z",
  "summary": "Violin Concerto #1 in a BWV 1041 by Johann Sebastian Bach \nLos Angeles Chamber Orchestra Jeffrey Kahane Margaret Batjer, violin \nBis 2309",
  "description": "sched_time: 68793500\nair_time: 69073000\nstack_pos: \ntitle: Violin Concerto #1 in a BWV 1041\nartist: Johann Sebastian Bach\ntrivia: \ncategory: MUB\ncart: 23TP\nintro: 0\nend: \nstation: KDFC-FM\nduration: 800531\nmedia_type: SONG\nmilliseconds_left: \nComposer: Johann Sebastian Bach\nComposer_Last: Bach, Johann Sebastian\nConductor: Jeffrey Kahane\nISRC: \nMM_Song_ID: 66187\nOrchestra: Los Angeles Chamber Orchestra\nRecord_Company: Bis\nSoloist: Margaret Batjer, violin\nSpine_Number: 2309\n",
  "start": {
    "dateTime": "2020-11-03T19:11:13-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "end": {
    "dateTime": "2020-11-03T19:24:33-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "extraInfo": {
    "sched_time": "68793500",
    "air_time": "69073000",
    "stack_pos": "",
    "title": "Violin Concerto #1 in a BWV 1041",
    "artist": "Johann Sebastian Bach",
    "trivia": "",
    "category": "MUB",
    "cart": "23TP",
    "intro": "0",
    "end": "",
    "station": "KDFC-FM",
    "duration": "800531",
    "media_type": "SONG",
    "milliseconds_left": "",
    "Composer": "Johann Sebastian Bach",
    "Composer_Last": "Bach, Johann Sebastian",
    "Conductor": "Jeffrey Kahane",
    "ISRC": "",
    "MM_Song_ID": "66187",
    "Orchestra": "Los Angeles Chamber Orchestra",
    "Record_Company": "Bis",
    "Soloist": "Margaret Batjer, violin",
    "Spine_Number": "2309"
  }
}
```

During an SF Symphony broadcast:

```json
{
  "id": "3aujsfkdob9958is0mprb93l8g_20201104T040000Z",
  "created": "2020-09-01T22:39:16.000Z",
  "updated": "2020-09-01T22:39:16.299Z",
  "summary": "The San Francisco Symphony Broadcast",
  "description": "sched_time:\nair_time:\nstack_pos: \ntitle: The San Francisco Symphony Broadcast\nartist: Hosted by Rik Malone\ntrivia: \ncategory: MUB\ncart:\nintro: 0\nend: \nstation: KDFC-FM\nduration:\nmedia_type: SONG\nmilliseconds_left: \nComposer: \nComposer_Last:\nConductor: \nISRC: \nMM_Song_ID:\nOrchestra: San Francisco Symphony\nRecord_Company: For more information, visit www.kdfc.com/sfsymphony\nSoloist: \nSpine_Number:",
  "start": {
    "dateTime": "2020-11-03T20:00:00-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "end": {
    "dateTime": "2020-11-03T21:30:00-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "recurringEventId": "3aujsfkdob9958is0mprb93l8g",
  "originalStartTime": {
    "dateTime": "2020-11-03T20:00:00-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "extraInfo": {
    "stack_pos": "",
    "title": "The San Francisco Symphony Broadcast",
    "artist": "Hosted by Rik Malone",
    "trivia": "",
    "category": "MUB",
    "intro": "0",
    "end": "",
    "station": "KDFC-FM",
    "media_type": "SONG",
    "milliseconds_left": "",
    "Composer": "",
    "Conductor": "",
    "ISRC": "",
    "Orchestra": "San Francisco Symphony",
    "Record_Company": "For more information, visit www.kdfc.com/sfsymphony",
    "Soloist": ""
  }
}
```

After the SF Symphony broadcast:

```json
{
  "id": "ou3p9ogj237c6linhv23mjv4nc",
  "created": "2020-11-04T05:54:27.000Z",
  "updated": "2020-11-04T05:54:27.304Z",
  "summary": "Concerto grosso #6 in D Op 3 HWV 317 by George Frideric Handel \nHandel and Haydn Society Christopher Hogwood  \nLOiseau 421729",
  "description": "sched_time: 77999000\nair_time: 78863000\nstack_pos: \ntitle: Concerto grosso #6 in D Op 3 HWV 317\nartist: George Frideric Handel\ntrivia: \ncategory: MUA\ncart: 00TJ\nintro: 180\nend: \nstation: KDFC-FM\nduration: 405000\nmedia_type: SONG\nmilliseconds_left: \nComposer: George Frideric Handel\nComposer_Last: Handel, George Frideric\nConductor: Christopher Hogwood\nISRC: \nMM_Song_ID: 2897\nOrchestra: Handel and Haydn Society\nRecord_Company: LOiseau\nSoloist: \nSpine_Number: 421729\n",
  "start": {
    "dateTime": "2020-11-03T21:54:23-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "end": {
    "dateTime": "2020-11-03T22:01:08-08:00",
    "timeZone": "America/Los_Angeles"
  },
  "extraInfo": {
    "sched_time": "77999000",
    "air_time": "78863000",
    "stack_pos": "",
    "title": "Concerto grosso #6 in D Op 3 HWV 317",
    "artist": "George Frideric Handel",
    "trivia": "",
    "category": "MUA",
    "cart": "00TJ",
    "intro": "180",
    "end": "",
    "station": "KDFC-FM",
    "duration": "405000",
    "media_type": "SONG",
    "milliseconds_left": "",
    "Composer": "George Frideric Handel",
    "Composer_Last": "Handel, George Frideric",
    "Conductor": "Christopher Hogwood",
    "ISRC": "",
    "MM_Song_ID": "2897",
    "Orchestra": "Handel and Haydn Society",
    "Record_Company": "LOiseau",
    "Soloist": "",
    "Spine_Number": "421729"
  }
}
```
